src/Microsoft.DotNet.Build.Tasks/GenerateCurrentVersion.cs

```diff
@@ -0,0 +1,96 @@
+        /// </summary>
+        public string ComparisonDate { get; set; }
+
+        /// <summary>
+        /// The Major Version that will be produced given a SeedDate.
+        /// </summary>
+        [Output]
+        public string GeneratedVersion { get; set; }
+
+        private const string DateFormat = "yyyy-MM-dd";
+        private CultureInfo enUS = new CultureInfo("en-US");
+        private int padding;
+
+        public override bool Execute()
+        {
+            if (string.IsNullOrEmpty(Padding))
+            {
+                Padding = "5";
+            }
+            bool isValidPadding = int.TryParse(Padding, out padding);
+            if (!isValidPadding || padding < 5)
+            {
+                Log.LogError("The specified padding '{0}' is invalid. Please pass in an intiger equal or greater than 5.", Padding);
+            }
+            DateTime date;
+            GeneratedVersion = string.Empty;
+            bool isValidDate = DateTime.TryParseExact(SeedDate, DateFormat, enUS, DateTimeStyles.AssumeLocal, out date);
+            if (!isValidDate)
+            {
+                Log.LogError("The seed date '{0}' is not valid. Please specify a date in the short format.(YYYY-MM-DD)", SeedDate);
+                return false;
+            }
+            GeneratedVersion = GetCurrentVersionForDate(date, ComparisonDate);
+            if (string.IsNullOrEmpty(GeneratedVersion))
+            {
+                Log.LogError("The date '{0}' is not valid. Please pass in a date after April 1996.", SeedDate);
+                return false;
+            }
+            return true;
+        }
+
+        public string GetCurrentVersionForDate(DateTime seedDate, string comparisonDate)
+        {
+            DateTime compareDate;
+            if (string.IsNullOrEmpty(comparisonDate))
+            {
+                /*
+                 * We need to ensure that our build numbers are higher that what we used to ship internal builds so this date
+                 * will make that possible.
+                 */
+                compareDate = new DateTime(1996, 4, 1);
+            }
+            else
+            {
+                bool isValidDate = DateTime.TryParseExact(comparisonDate, DateFormat, enUS, DateTimeStyles.AssumeLocal, out compareDate);
+                if (!isValidDate)
+                {
+                    Log.LogError("The comparison date '{0}' is not valid. Please specify a date in the short format.(YYYY-MM-DD)", comparisonDate);
+                }
+            }
+            int months = (seedDate.Year - compareDate.Year) * 12 + seedDate.Month - compareDate.Month;
+            if (months > 0) //only allow dates after comparedate
+            {
+                return string.Format("{0}{1}", months.ToString("D" + (padding-2)), seedDate.Day.ToString("D2"));
+            }
+            return string.Empty;
```

Comment:
Actually I just noticed you error on empty above so you can ignore this comment. I'm fine with it being an error.